packages/netlify-cms-core/src/formats/yaml.js

```diff
@@ -1,49 +1,38 @@
-import yaml from 'js-yaml';
-import moment from 'moment';
-import AssetProxy from 'ValueObjects/AssetProxy';
+import yaml from 'yaml';
 import { sortKeys } from './helpers';
 
-const MomentType = new yaml.Type('date', {
```

Comment:
Couldn't get `moment.isMoment(value)`  and `value instanceof AssetProxy` to evaluate to `true`.
We can add custom types to `yaml`, but looks like it is not required. @erquhart  thoughts?

Reply:
The date widget used to pass a moment instance if no format was configured, but that doesn't seem possible anymore:
https://github.com/netlify/netlify-cms/blob/master/packages/netlify-cms-widget-date/src/DateControl.js#L94

Maybe something similar happened with `AssetProxy`, but ensuring that may be more complex than just checking the change handler for the image widget. If you feel confident that an instance of `AssetProxy` won't end up as a field value in the redux store, we should be good to remove the related logic here.

Reply:
Looks like the image widget(s) pass strings/arrays of strings:
https://github.com/netlify/netlify-cms/blob/2fe9ab72f38a1659011979069cfe196a2c35cff3/packages/netlify-cms-widget-file/src/withFileControl.js#L226
https://github.com/netlify/netlify-cms/blob/2fe9ab72f38a1659011979069cfe196a2c35cff3/packages/netlify-cms-core/src/actions/mediaLibrary.ts#L112